### PR TITLE
recast std::bad_alloc as a Python exception

### LINKED ISSUE #111: Put useful error message in when large hash tables fail to alloc.

### PR DESCRIPTION
**Pull request by mr-c**
Will fix #111

### CODE REVIEW

**Comment by mr-c** [comment]
- [ ] Is it mergable?
- [ ] Did it pass the tests?
- [ ] If it introduces new functionality in scripts/ is it tested?
  Check for code coverage with `make clean diff-cover`
- [ ] Is it well formatted? Look at `make pep8`, `make diff_pylint_report`,
  `make cppcheck`, and `make doc` output. Use `make format` and manual
  fixing as needed.
- [ ] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [ ] Is it documented in the ChangeLog?
  http://en.wikipedia.org/wiki/Changelog#Format
- [ ] Was a spellchecker run on the source code and documentation after
  changes were made?

**Comment by mr-c** [comment]
Jenkins, test this please

**Comment by mr-c** [comment]
@luizirber @ctb @camillescott Ready for review and merge

File: Makefile [hunk 1]
```diff
@@ -162,7 +168,7 @@ diff-cover.html: coverage-gcovr.xml coverage.xml
 		--html-report diff-cover.html
 
 nosetests.xml: FORCE
-	./setup.py nosetests --with-xunit
+	./setup.py nosetests --with-xunit --attr ${TESTATTR}
```
Comment (ctb) [inline Makefile]: does this mean we no longer need setup.cfg?

Comment (mr-c) [inline Makefile]: setup.cfg provides other default values and acts as a fallback if the Setuptools nosetests command is run outside of the Makefile

**Pull request merged**